src/posts/lucky_0.23.cr

```diff
@@ -0,0 +1,219 @@
+
+    ## Quick Updates
+
+    ### `mount` is now `m`
+
+    We have deprecated the use of `mount` in favor of a shorter method `m`. The structure of this
+    is a little bit different.
+
+    ```crystal
+    # Before
+    mount MyComponent.new(arg: 1)
+
+    # After
+    m MyComponent, arg: 1
+    ```
+
+    As you can see, you still have all the flexibility, but in a bit more compact version!
+
+    ### Params
+
+    When you need to access a param value, you'll use `params.get(:key)`, but now, the blankspace
+    will be stripped automatically. If you need to get the untouched value, you can use `params.get_raw`.
+
+    ### Memoize 2.0
+
+    The `Memoizable` module got a nice little upgrade. It now supports arguments, as well as
+    `false` and `nil` values. 
+
+    ```crystal
+    memoize user(user_id : Int64) : User?
+      UserQuery.new.find(user_id)
+    end
+    ```
+
+    Now when you call this method, it will run the first time, and return the value even if that value is
+    `nil`. If you need access to the uncached version you can use the `__uncached` version of the method.
+
+    ```crystal
+    # run the query once, and always return that value
+    user(4)
+
+    # always run the query and return the value
+    user__uncached(4)
+    ```
+
+    ### Security and more
+    
+    Some of the HTML page helpers like `highlight`, and `simple_format` would allow for unescaped HTML.
+    This could potentially be a security issue, so that's been fixed. All of these methods will escape the
+    HTML by default now. If you need the unescaped version, you can pass `escape: false`.
+
+    ```crystal
+    highlight("<p>This is a beautiful morning, but also a beautiful day</p>", "beautiful", escape: false)
+    ```
+
+    We've also fixed a few issues related to cookies.
+    * You can now set a `samesite` default.
+    * Exceptions aren't raised when trying to delete a cookie that doesn't exist
+    * Cookies are written at the correct time
+
+    ## Routing Updates
+
+    We have added a new `route_prefix` macro that alllows you to prefix your routes with
+    some path. This really helps with APIs where you may want to prefix all your routes with
+    something like `/api/v1`.
+
+    ```crystal
+    abstract class ApiAction < Lucky::Action
+      accepted_formats [:json], default: :json
+
+      route_prefix "/api/v1"
+    end
+
+    class Posts::Index < ApiAction
+      # GET /api/v1/posts`
+      get "/posts" do
+        #...
+      end
+    end
+    ```
+
+    Sometimes you may have a route with optional path params. For example, imagine a list of
+    blog posts that can be optionally filtered by appending the year, or year and month.
+
+    ```crystal
+    class Posts::Index < BrowserAction
+      # Matches GET /posts
+      # GET /posts/2020
+      # GET /posts/2020/07
+      get "/posts/?:year/?:month" do
+        # ...
+      end
+    end
```

Comment:
Does this actually work? I think I added it to the router, but I don't think there is support in Lucky for extracting those parts and making a nilable param. So I *think* this should be removed until we add support to Lucky::Routable

Reply:
oh... I didn't even check. I assumed it did 😅  yeah, I'll take that out.